entrypoint.sh

```diff
@@ -12,6 +12,13 @@ if echo ${@} | grep -q "cloud-provider=azure"; then
   fi
 fi
 
+if [ "$1" = "kube-proxy" ]; then
+  if echo ${@} | grep -v "hostname-override"; then
+    hostname=$(curl "http://169.254.169.254/latest/meta-data/hostname")
```

Comment:
what is this ip it is pinned to?

Reply:
It's the instance metadata service ip, https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instancedata-data-retrieval.html

Reply:
would this still be accessible in an airgap setup?

Reply:
Yep, I haven't tested this but the docs call it out specifically - This IPv4 address is a link-local address and it is valid only from the instance.